components/automate-ui/src/app/pages/+compliance/+reporting/reporting.component.spec.ts

```diff
@@ -61,51 +61,26 @@ describe('ReportingComponent', () => {
 
   describe('sets availableFilterTypes', () => {
     it('to include profile, platform, environment, node, and control', () => {
-      expect(component.availableFilterTypes).toEqual([
-        {
-          'name': 'profile',
-          'title': 'Profile',
-          'description': 'Add the name or ID to filter this report against a profile',
-          'placeholder': 'Name or ID'
-        },
-        {
-          'name': 'node',
-          'title': 'Node',
-          'description':
-          'Add the name, ID, or hostname to filter this report against a specific node',
-          'placeholder': 'Name, ID, or hostname'
-        },
-        {
-          'name': 'platform',
-          'title': 'Platform',
-          'description': 'Add the name to filter this report to a specific platform',
-          'placeholder': 'Name'
-        },
-        {
-          'name': 'environment',
-          'title': 'Environment',
-          'description': 'Add the name to filter this report to a specific environment',
-          'placeholder': 'Name'
-        },
-        {
-          'name': 'control',
-          'title': 'Control',
-          'description': 'Add the title to filter this report against a control',
-          'placeholder': 'Title'
-        },
-        {
-          'name': 'role',
-          'title': 'Role',
-          'description': 'Add the role to filter this report to a specific role',
-          'placeholder': 'Role'
-        },
-        {
-          'name': 'recipe',
-          'title': 'Recipe',
-          'description': 'Add the recipe to filter this report to a specific recipe',
-          'placeholder': 'Recipe'
-        }
-      ]);
+      const availableFilterTypesNames = component.availableFilterTypes.map( type => type.name);
+
+      const expected = [
+        'chef_server', 'chef_tags', 'control', 'environment', 'node', 'organization', 'platform',
+        'policy_group', 'policy_name', 'profile', 'recipe', 'role'];
+
+      expect(availableFilterTypesNames.length).toBe(expected.length);
+
+      // Each expected type is in the availableFilterTypesNames
+      expected.forEach(expectedType =>
+          expect(availableFilterTypesNames.indexOf(expectedType)).toBeGreaterThanOrEqual(0));
```

Comment:
What does this do that testing for equality doesn't? This would allow `availableFilterTypesNames` to have elements not in `expected`, is that necessary behavior?

Reply:
The length check above would preclude my concern above, but I'm still not sure what this is doing that an equality check wouldn't.

Reply:
I did not want to use equals because that would dictate the order. The order could be different than the `expected` array because the titles are what is being used to sort. 
I think it is correct that `availableFilterTypesNames` can have values not in `expected`. I would not want to fail when someone adds a new element to the search bar.

Reply:
If there isn't an assertion for "same elements in any order" you could do something like `expect(list1.sort()).toEqual(list2.sort())`.

Reply:
I could not find an assertion for "same elements in any order" or ElementMatch in jasmine. 

I will try out your suggested code.

Reply:
That works `expect(expected.sort()).toEqual(availableFilterTypesNames.sort());`. 
Thanks. I am pushing it up now.